### ISSUE TITLE
update/patch resource name

### ISSUE THREAD

**Issue by JeanMertz** [kubernetes/kubernetes#13629]
I'd like to update a replication controller's name using `kubectl`, but am unable to do so.

Basically, I want to mimic the `rolling-update` command where it creates a temporary rc name and then renames it.

There are the steps I'd like to take:

```
kubectl rolling-update app app-tmp -f app-rc.json
kubectl patch rc app-tmp -p '{ "metadata": { "name": "app" } }'
```

And the error I am getting:

```
Error from server: the name of the object (app) does not match the name on the URL (app-tmp)
```

**Comment by JeanMertz** [comment]
I've found the documentation regarding a rolling update. This explains that after the update, a new RC is created, and the old one removed.

So I guess the "correct" description would be

```
kubectl rolling-update app app-tmp -f app-rc.json
kubectl get rc app-tmp -o json | jq '.metadata.name |= "app"' | kubectl create -f-
kubectl delete rc app-tmp
```

right?

This seems rather complicated for a resource rename.

The reason for this rename is simple: to keep documentation up-to-date with the real-world scenario. That is, if our documentation states to do `kubectl get app` to get the app status, there should be no surprise when running the same command a while later, when several updates have occured and the command should have been `kubectl get app-v3`.

**Comment by jlowdermilk** [comment]
I think from an api standpoint, we can't allow changing the name/id of a resource as an "update" action. If you want to change the name of a resource, you have to delete and recreate.

cc @bgrant0607

**Comment by JeanMertz** [comment]
Well, That depends on how names are used

To me, when working with Kunernetes, it feels like

* labels are used for cross-resource referencing
* internal IDs are used for required hard references
* names are used as human readable identifiers, so that two people in a meeting understand what they mean when they reference the "mailing RC".

Given this, keeping names equal and readable throughout the lifecycle of a resource seems very much like a desirable feature to me.

**Comment by bgrant0607** [comment]
@JeanMertz This overlaps heavily with another issue you filed -- #8827.

The deployment API, which I believe is what you want, is in progress. Proposal is here:
https://github.com/kubernetes/kubernetes/blob/master/docs/proposals/deployment.md

As @jlowdermilk said, names will not be made updatable. Please see https://github.com/kubernetes/kubernetes/blob/master/docs/design/identifiers.md#general-design

I suggest using labels instead for what you're trying to do:
```
kubectl get rc -lapp=myapp
```

**Comment by okoreniuk** [comment]
> names will not be made updatable.
Same time http://kubernetes.io/docs/user-guide/rolling-updates/
If no NEW_NAME is specified, a new replication controller is created with a temporary name. Once the rollout is complete, the old controller is deleted, and the new controller is **## updated** to use the original name.
Are you going to change this policy?
